### PR TITLE
add contribute/design-guidelines

### LINKED ISSUE #1946: PatternFly design doc contribution guidelines: Process
Document the design doc contribution process on PatternFly. Add the information to the [PatternFly Contribute page.](https://www.patternfly.org/v4/contribute/about) 

Incorporate information from these internal docs: 

- **[Contributing PatternFly Design Guidelines]**(https://docs.google.com/document/d/1mVg5NcXkk0uk3kftOv7pvag991oL1XNvti2z5ddfjH4/edit?usp=sharing) 

- **[Contributing to patternfly.org]**(https://docs.google.com/document/d/1kaDqOZel_zEuLbPNKftl84SIxnA8doEAaxdOE5Y__m0/edit?usp=sharing) 

This information will focus on the steps involved in contributing, not on the docs' written content. We'll be creating a new page on the PF content style guide for writing design docs. We'll LINK to that page from this Contribute page.  

Issue for written content info, #1947

### PR DESCRIPTION
**Pull request by redallen**
Credit to @japhilli-RH and @abigaeljamie for the content. Taken from their [Google drive folder](https://drive.google.com/drive/folders/1_NcxCJWNvPKwqAo2OFQkB7DHBnpAa9pH?usp=sharing).

Closes #1946

### CODE REVIEW

File: packages/v4/src/content/contribute/design-guidelines/design-guidelines.md
Comment (mmenestr) [inline packages/v4/src/content/contribute/design-guidelines/design-guidelines.md:190]: Link to my GitHub is wrong -- it says "mmenestret" but should just be "/mmenestr" just like the tagged text

**Review by mmenestr — CHANGES_REQUESTED** [review]

File: packages/v4/src/content/contribute/design-guidelines/design-guidelines.md
Comment (redallen) [reply packages/v4/src/content/contribute/design-guidelines/design-guidelines.md]: Fixed!

**Review by mmenestr — APPROVED** [review]
Looks good, but don't merge till @gdoyle1 's had a chance to look at it!

**Comment by gdoyle1** [comment]
@redallen some of the steps on the TOC don't link correctly - step 2 and step 3 and then step 8 and step 9 will highlight the wrong ones

**Review by gdoyle1 — CHANGES_REQUESTED** [review]
@redallen Noticed a couple things...
1. Some of the steps on the TOC don't link correctly - step 2 and step 3 and then step 8 and step 9 will highlight the wrong ones
2. We do have a visual design request google form out there, @mceledonia is that something we would want to surface on the site rather than sending people directly to me?
Screen Shot 2020-09-08 at 6 53 24 PM

**Review by abigaeljamie — DISMISSED** [review]
Thank you, @redallen !

**Comment by redallen** [comment]
@gdoyle1 The TOC scrolling should be fixed. It was caused by incorrect offsets (I was using the title header as an offset instead of the masthead's consistent 76px) and images loading (as they load in the height of page changes).

**Comment by mceledonia** [comment]
@gdoyle1 @redallen We might want to remove that section completely. We do have the visual design request form, but I believe we want to keep that to internal Red Hat / product teams. I think if a community contribution makes it past the proposal phase we will be already incorporating it into our planning and assigning UX/Visual aid in that way.

**Comment by redallen** [comment]
@gdoyle1 Per @mceledonia 's suggestion I've removed that paragraph entirely.

@mmenestr @abigaeljamie @gdoyle1 Please rereview!

**Review by gdoyle1 — CHANGES_REQUESTED** [review]
Couple other things I found @redallen ! ...
1. "When creating high-fidelity mockups, you should first look for existing examples in the **PF4-Components Sketch file**."
"PF4-Components Sketch file" should be "PatternFly template file" and link here: https://www.sketch.com/s/729c2eee-e8b6-4fcd-8a79-f6faa8c30f89.
2. The "view source on GitHub" link is broken

**Comment by redallen** [comment]
@gdoyle1

1. Done
2. The "View source on Github" link will work after the PR is merged and the file is in the master branch on Github.

**Review by gdoyle1 — APPROVED** [review]
All set now, thank you @redallen !!

**Pull request merged**
